### DISCUSSION TITLE
GitHub Actions workflow not triggered with path/**

### DISCUSSION THREAD

**Discussion by SamChou19815** [community/community#25285] [Actions]
<p>I define a workflow that will run on push to master if anything in sam-highlighter directory changes:</p>
<p><a href="https://github.com/SamChou19815/website/blob/master/.github/workflows/cd-blog.yml" rel="nofollow noopener">https://github.com/SamChou19815/website/blob/master/.github/workflows/cd-blog.yml</a></p>
<p>However, it is not triggered on master run in the latest commit, and this is the relevant commit: <a href="https://github.com/SamChou19815/website/commit/dbdc6ab5145edc407cf6b06c08faeca5558e75ba" rel="nofollow noopener">https://github.com/SamChou19815/website/commit/dbdc6ab5145edc407cf6b06c08faeca5558e75ba</a></p>
<p>This is my repository:</p>
<p><a href="https://github.com/SamChou19815/website" rel="nofollow noopener">https://github.com/SamChou19815/website</a></p>

**Comment by lee-dohm** [comment]
<p>I don’t see the workflow you linked to anymore. Is this still an issue? If so, can you tell me which workflow is the one you’re having a problem with?</p>
<p>Thanks.</p>

**Comment by SamChou19815** [comment]
<p>Here is a more recent commit: <a href="https://github.com/SamChou19815/website/commit/1a1643a643b764337626b8a16d05fe56d8bae913" rel="nofollow noopener">https://github.com/SamChou19815/website/commit/1a1643a643b764337626b8a16d05fe56d8bae913</a></p>
<p>and workflow files at that point:</p>
<p><a href="https://github.com/SamChou19815/website/tree/1a1643a643b764337626b8a16d05fe56d8bae913/.github/workflows" rel="nofollow noopener">https://github.com/SamChou19815/website/tree/1a1643a643b764337626b8a16d05fe56d8bae913/.github/workflows</a></p>
<p>Since I made a change in the main-site-frontend folder in master, I would expect this workflow to run based on the path config: <a href="https://github.com/SamChou19815/website/blob/1a1643a643b764337626b8a16d05fe56d8bae913/.github/workflows/generated-cd-main-site-frontend.yml" rel="nofollow noopener">https://github.com/SamChou19815/website/blob/1a1643a643b764337626b8a16d05fe56d8bae913/.github/workflows/generated-cd-main-site-frontend.yml</a> but it’s not run.</p>

**Comment by JesseTG** [comment]
<p>You expect that pattern to cover arbitrarily-nested directories, right?  I tried it out earlier, and I don’t believe that’s supported right now.</p>

**Comment by jeremyepling** [comment]
<pre><code>'main-site-frontend/**'
</code></pre>
<p>Try changing the above to</p>
<pre><code>'main-site-frontend/*'
</code></pre>

**Comment by RobinTTY** [comment]
<p>I have tested using only one asterisk and it gives the same result as two asterisks. Only files directly in the folder will trigger the action but not a file which is nested, for instance: main-site-frontend/another-folder/changed-file.txt will not trigger the action</p>

**Comment by svalluru-aai** [comment]
<p>I think that the pattern is using Go Path, which does not seem to have any support for nested directories.</p>
<p>I checked the Go path discussions and it seems to have gone anywhere to being supported. This fundamentally becomes a maintenance issue. I had to use a bunch of ‘path/*’, ‘path/*/*’, etc… pasted over and over again to get the result I wanted. However if someone were to add one more nested folder than my configuration and forgets to change the workflow file, it will not get triggered.</p>

**Comment by greenstatic** [comment]
<p><span class="mention">@svalluru-aaiis</span> right, Go path.Match does not support matching nested directories.</p>
<p>On <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#example-matching-paths-in-nested-directories" rel="nofollow noopener">GitHub’s docs there is a section</a> on how to support sub-projects by listing as many nested path’s as possible which is a downright hack.</p>
<p>This feature is crutial in supporting monorepos and would be amazing if it worked out of the box for such usecases.</p>
<p>Would it be possible to switch from path.Match to something like regex? This would enable much more versitile rules and the Go standard library already comes with regex in the standard library.</p>

**Comment by chrispat** [comment, marked as answer]
<p>We have a new syntax for branch, tag and path filtering <a href="https://github.blog/changelog/2019-09-30-github-actions-event-filtering-updates/" rel="nofollow noopener">https://github.blog/changelog/2019-09-30-github-actions-event-filtering-updates/</a>.</p>

**Comment by riker09** [comment]
<p>I’m currently evaluating moving more of our CI/CD workload to GitHub Actions. Our project is a Monorepo with more than 100 nested projects. It would be extremly helpful if I could add </p>
<pre><code>**/*.js
</code></pre>
<p>to match <em>ALL</em> JavaScript files in all subdirectories.</p>
<p>Right now I’m using this instead:</p>
<pre><code>*.js
*/*.js
*/*/*.js
*/*/*/*.js
*/*/*/*/*.js
*/*/*/*/*/*.js
</code></pre>
<p>This will trigger when changes happen to any JS file nested 5 levels deep. The workflow file gets very hard to read, though.</p>
<p>Any idea about the timeline when the feature will land?</p>

**Comment by chrispat** [comment]
<p>We are working on an update to the path matching that will support the more standandard globbing patterns.  Please keep an eye on the GitHub changelog for the update.</p>

**Comment by unrevised6419** [comment]
<p>Here it is!!!</p>
<p><a href="https://github.blog/changelog/2019-09-30-github-actions-event-filtering-updates/" rel="nofollow noopener">https://github.blog/changelog/2019-09-30-github-actions-event-filtering-updates/</a></p>

**Comment by mrtrkmn** [comment]
<p>Unfortunately, when I type <code>paths-ignore</code>, Github is complaining about it, not working.</p>
<p>Sorry, it is working but auto-suggestion of Github is complaining about the syntax that’s why I though it is not valid. I am so sorry :frowning:</p>

**Comment by fransf-wtax** [comment]
<p>How can we debug path matching? Currently if the workflow doesn’t start, we’re just left staring at the Actions screen with one big question: WHY?<br>
Can we somewhere get a log of how the changed files in a commit are matched against the paths?</p>

**Comment by nathanielford** [comment]
<p>Or, for that matter, the inverse - why did a workflow run when it should not have? It would be nice to see a list of files, filters they matched and the workflow decision.</p>
